numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -2331,6 +2404,38 @@ npyiter_ass_subscript(NewNpyArrayIterObject *self, PyObject *op,
     return -1;
 }
 
+static PyObject *
+npyiter_enter(NewNpyArrayIterObject *self)
+{
+    if (self->iter == NULL) {
+        PyErr_SetString(PyExc_ValueError, "operation on non-initialized iterator");
```

Comment:
Can this ever happen? If not, should probably be a RuntimeError

Reply:
fixed